src/frontend/src/binder/set_expr.rs

```diff
@@ -194,6 +218,85 @@ impl Binder {
         Ok(())
     }
 
+    /// Check the corresponding specification of the set operation.
+    /// Returns the corresponding column index of the left and right side.
+    fn corresponding(
+        &self,
+        left: &BoundSetExpr,
+        right: &BoundSetExpr,
+        corresponding: Corresponding,
+    ) -> Result<(ColIndexMapping, ColIndexMapping)> {
+        let check_duplicate_name = |set_expr: &BoundSetExpr| {
+            let mut name2idx = HashMap::new();
+            for (idx, field) in set_expr.schema().fields.iter().enumerate() {
+                if name2idx.insert(field.name.clone(), idx).is_some() {
+                    return Err(ErrorCode::InvalidInputSyntax(format!(
+                        "duplicated column name `{}` in a select list of a set operation",
+                        field.name,
+                    )));
+                }
+            }
+            Ok(name2idx)
+        };
+
+        // Within the columns of both side, the same <column name> shall not
+        // be specified more than once.
+        let name2idx_l = check_duplicate_name(left)?;
+        let name2idx_r = check_duplicate_name(right)?;
+
+        let mut corresponding_col_idx_l = vec![];
+        let mut corresponding_col_idx_r = vec![];
+
+        if let Some(column_list) = corresponding.column_list() {
+            // The select list of the corresponding set operation should be in the order of <corresponding column list>
+            for column in column_list {
+                let col_name = column.real_value();
+                if let Some(idx_l) = name2idx_l.get(&col_name)
+                    && let Some(idx_r) = name2idx_r.get(&col_name)
+                {
+                    corresponding_col_idx_l.push(*idx_l);
+                    corresponding_col_idx_r.push(*idx_r);
+                } else {
+                    return Err(ErrorCode::InvalidInputSyntax(format!(
+                        "Every <column name> in the <corresponding column list> shall be a \
+                        <column name> of both left and right side. Missing column: `{}`",
```

Comment:
1. "corresponding column list" seems a little verbose. "corresponding clause"?
2. Why using `<>`? This seems unnecessary and inconsistent with our other error messages.

Reply:
And better to indicate which side is missing.

Reply:
> And better to indicate which side is missing.

Ditto